lottie/src/main/java/com/airbnb/lottie/network/NetworkFetcher.java

```diff
@@ -84,36 +84,47 @@ private LottieComposition fetchFromCache() {
 
   @WorkerThread
   private LottieResult fetchFromNetworkInternal() throws IOException {
-    L.debug( "Fetching " + url);
-    HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
+    L.logger.debug("Fetching " + url);
 
+    HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
     connection.setRequestMethod("GET");
     connection.connect();
 
     if (connection.getErrorStream() != null || connection.getResponseCode() != HttpURLConnection.HTTP_OK) {
+
+      int responseCode = connection.getResponseCode();
       BufferedReader r = new BufferedReader(new InputStreamReader(connection.getErrorStream()));
       StringBuilder error = new StringBuilder();
       String line;
       while ((line = r.readLine()) != null) {
         error.append(line).append('\n');
       }
+
+      // Be sure to clean resources
+      try {
```

Comment:
You want to wrap the actual connection with try and clean up in finally.

Reply:
Do you mean something like:

`

    HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
    connection.setRequestMethod("GET");
    try {
      connection.connect();

      if (connection.getErrorStream() != null || connection.getResponseCode() != HttpURLConnection.HTTP_OK) {

        int responseCode = connection.getResponseCode();
        BufferedReader r = new BufferedReader(new InputStreamReader(connection.getErrorStream()));
        StringBuilder error = new StringBuilder();
        String line;

        try {
          while ((line = r.readLine()) != null) {
            error.append(line).append('\n');
          }
        } catch (Exception e) {
          throw e;
        } finally{
          r.close();
        }

        return new LottieResult<>(new IllegalArgumentException("Unable to fetch " + url + ". Failed with " +
            responseCode + "\n" + error));
      }
    } catch (Exception e) {
      return new LottieResult<>(e);
    } finally {
      connection.disconnect();
    }`

Reply:
yup!

Reply:
Thanks for feedback, fix pushed!